apps/omg_watcher_rpc/test/omg_watcher_rpc/web/validators/typed_data_signed_test.exs

```diff
@@ -0,0 +1,116 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherRPC.Web.Validators.TypedDataSignedTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: true
+
+  alias OMG.State.Transaction
+  alias OMG.TestHelper
+  alias OMG.Utils.HttpRPC.Encoding
+  alias OMG.Utxo
+  alias OMG.WatcherRPC.Web.Validator.TypedDataSigned
+
+  require Utxo
+
+  @eth OMG.Eth.RootChain.eth_pseudo_address()
+  @other_token <<127::160>>
+  @eth_hex Encoding.to_hex(@eth)
+  @token_hex Encoding.to_hex(@other_token)
+  @alice TestHelper.generate_entity()
+  @bob TestHelper.generate_entity()
+  @ari_network_address "44DE0EC539B8C4A4B530C78620FE8320167F2F74" |> Base.decode16!()
+  @eip_domain %{
+    name: "OMG Network",
+    version: "1",
+    salt: <<0::256>>,
+    verifyingContract: @ari_network_address
+  }
+
+  defp get_message do
```

Comment:
typed_data_request could be common

Reply:
Yes, these structures will be defined (and moved out of test) in next PR